### DISCUSSION TITLE
CustomCmd usage

### DISCUSSION THREAD

**Discussion by Klemsus** [faaxm/spix#37] [Q&A] [answered]
How method `command(command, payload)` works? 
I want to execute some command line tool on remote machine where Qt app with Spix is running.

Also I couldn't find any call to `TestServer::setGenericCommandHandler`. Does it mean that I should assign handler by myself first before running `command` remotely?

**Comment by faaxm** [comment, marked as answer]
The `command` is not a command line command, but rather a block of code that you can trigger from the test.

So as you guessed, you would call `TestServer::setGenericCommandHandler` somewhere in your application (where you also set up the test server) and register some function with it. This function could then be triggered via the `command` call in a test.

Mostly this can be used if you control the test from a script and want to have an easy way to reset or change the state of your application without going through the GUI.
If you want to run a commandline command, you would have to write the code that calls the commandline command yourself, but could then wrap it in a `CommandHandler` to call it with the `command` call from your test script.
